src/modules/MouseUtils/MouseHighlighter/dllmain.cpp

```diff
@@ -0,0 +1,316 @@
+        init_settings();
+    };
+
+    // Destroy the powertoy and free memory
+    virtual void destroy() override
+    {
+        delete this;
+    }
+
+    // Return the localized display name of the powertoy
+    virtual const wchar_t* get_name() override
+    {
+        return MODULE_NAME;
+    }
+
+    // Return the non localized key of the powertoy, this will be cached by the runner
+    virtual const wchar_t* get_key() override
+    {
+        return MODULE_NAME;
+    }
+
+    // Return JSON with the configuration options.
+    virtual bool get_config(wchar_t* buffer, int* buffer_size) override
+    {
+        HINSTANCE hinstance = reinterpret_cast<HINSTANCE>(&__ImageBase);
+        PowerToysSettings::Settings settings(hinstance, get_name());
+        return settings.serialize_to_buffer(buffer, buffer_size);
+    }
+
+    // Signal from the Settings editor to call a custom action.
+    // This can be used to spawn more complex editors.
+    virtual void call_custom_action(const wchar_t* action) override
+    {
+    }
+
+    // Called by the runner to pass the updated settings values as a serialized JSON.
+    virtual void set_config(const wchar_t* config) override
+    {
+        try
+        {
+            // Parse the input JSON string.
+            PowerToysSettings::PowerToyValues values =
+                PowerToysSettings::PowerToyValues::from_json_string(config, get_key());
+
+            parse_settings(values);
+
+            MouseHighlighterApplySettings(m_highlightSettings);
+        }
+        catch (std::exception&)
+        {
+            // Improper JSON.
+        }
+    }
+
+    // Enable the powertoy
+    virtual void enable()
+    {
+        m_enabled = true;
+        Trace::EnableMouseHighlighter(true);
+        std::thread([=]() { MouseHighlighterMain(m_hModule, m_highlightSettings); }).detach();
+    }
+
+    // Disable the powertoy
+    virtual void disable()
+    {
+        m_enabled = false;
+        Trace::EnableMouseHighlighter(false);
+        MouseHighlighterDisable();
+    }
+
+    // Returns if the powertoys is enabled
+    virtual bool is_enabled() override
+    {
+        return m_enabled;
+    }
+
+    virtual std::optional<HotkeyEx> GetHotkeyEx() override
+    {
+        return m_hotkey;
+    }
+
+    virtual void OnHotkeyEx() override
+    {
+        MouseHighlighterSwitch();
+    }
+
+    // Load the settings file.
+    void init_settings()
+    {
+        try
+        {
+            // Load and parse the settings file for this PowerToy.
+            PowerToysSettings::PowerToyValues settings =
+                PowerToysSettings::PowerToyValues::load_from_settings_file(MouseHighlighter::get_key());
+            parse_settings(settings);
+        }
+        catch (std::exception&)
+        {
+            // Error while loading from the settings file. Let default values stay as they are.
+        }
+    }
+
+    void parse_settings(PowerToysSettings::PowerToyValues& settings)
```

Comment:
Shouldn't `PowerToyValues` be const?

Reply:
`get_raw_json` is not const. (It's possible it could be, but that's out of the scope of this PR)